app/controllers/concerns/next_link_header.rb

```diff
@@ -5,12 +5,21 @@ module NextLinkHeader
 
   def set_next_link_header_using_changed_since_or_last_object(last_object,
                                                               params = {})
+    next_url_params = {}
+    next_url_params[:changed_since] = params[:changed_since]
+    next_url_params[:per_page] = params[:per_page]
+
     if last_object.present?
-      params[:changed_since] =
+      next_url_params[:changed_since] =
         incremental_load_timestamp_format last_object.changed_at
     end
 
-    response.headers['Link'] = "#{url_for(params: params)}; rel=\"next\""
+    if params[:recruitment_year].nil?
+      response.headers['Link'] = "#{url_for(recruitment_year: Settings.current_recruitment_cycle, params: next_url_params)}; rel=\"next\""
+      return
+    end
```

Comment:
I find the return here a little confusing, and also had to check to see if the string value for the Link header changes between the two commands. Can I suggest:

```ruby
cycle_year = params.fetch(:recruitment_year, Settings.current_recruitment_cycle)

response.headers['Link'] = "#{url_for(recruitment_year: cycle_year, params: next_url_params)}; rel=\"next\""
```

Reply:
Your suggestion did not return the correct cycle when params[:recruitment_year] was nil so i've added the following:
```

  if params[:recruitment_year].nil?
      cycle_year = Settings.current_recruitment_cycle
      response.headers['Link'] = "#{url_for(recruitment_year: cycle_year, params: next_url_params)}; rel=\"next\""
    else
      response.headers['Link'] = "#{url_for(recruitment_year: params[:recruitment_year], params: next_url_params)}; rel=\"next\""
    end

```